### PR TITLE
feat(console): support icon in button

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

* support icon in button
* add icon `More` and add `More Button` into api-resource-details page

## Linear Issue Reference

None.

## Testing

image

### CODE REVIEW

**Review by wangsijie — COMMENTED** [review]
What about just merge `icon` and `title` to `children`? @logto-io/eng 
```
<Button><MoreIcon /> {t('more')} </Button>
```

File: packages/console/src/components/Button/index.module.scss [hunk 1]
```diff
@@ -4,10 +4,15 @@
...
   outline: none;
   border-radius: _.unit(2);
+  vertical-align: middle;
   font: var(--font-button);
   transition: background 0.2s ease-in-out;
   @include _.text-ellipsis;
 
+  > svg {
+    vertical-align: middle;
+  }
```
Comment (demonzoo) [inline packages/console/src/components/Button/index.module.scss:14]: Better use flexbox?
```css
display: flex;
align-items: center;
```

**Review by demonzoo — APPROVED** [review]

**Comment by gao-sun** [comment]
@wangsijie i think `icon={}` would be better since it's clear

File: packages/console/src/components/Button/index.tsx [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
+import React, { HTMLProps, ReactNode } from 'react';
 import { useTranslation } from 'react-i18next';
 
 import * as styles from './index.module.scss';
 
-type Props = Omit<HTMLProps<HTMLButtonElement>, 'type' | 'size' | 'title'> & {
+type Props = Omit<HTMLProps<HTMLButtonElement>, 'type' | 'size' | 'icon' | 'title'> & {
   htmlType?: 'button' | 'submit' | 'reset';
-  title: I18nKey;
+  icon?: ReactNode;
```
Comment (wangsijie) [inline packages/console/src/components/Button/index.tsx:10]: Maybe we can defined a `Icon` component with props like size, can do it later or take a discussion.

**Review by wangsijie — APPROVED** [review]

File: packages/console/src/components/Button/index.tsx (hunk 2 above)
Comment (gao-sun) [reply packages/console/src/components/Button/index.tsx:10]: let's add this into discussion

File: packages/console/src/components/Button/index.tsx [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
+  title?: I18nKey;
```
Comment (simeng-li) [inline packages/console/src/components/Button/index.tsx:11]: should the title be required?  Feels like IconButton should be an explicit component.

Comment (xiaoyijun) [reply packages/console/src/components/Button/index.tsx:11]: Will be discussed in LOG-1888.

File: packages/console/src/components/Button/index.tsx (hunk 2 above)
Comment (xiaoyijun) [reply packages/console/src/components/Button/index.tsx:10]: LOG-1888

File: packages/console/src/components/Button/index.tsx (hunk 3 above)
Comment (wangsijie) [reply packages/console/src/components/Button/index.tsx:11]: I agree with @simeng-li

**Comment by wangsijie** [comment]
Superseded by #393

**Pull request closed without merging**